### ISSUE TITLE
`stripplot` and `swarmplot` fail on empty data when `palette` is set

### ISSUE THREAD

**Issue by MaozGelbart** [mwaskom/seaborn#3118]
You can do the following and get an empty plot:
```python
sns.stripplot(x=[], y=[], hue=[], palette=None)
```
But providing a palette raises an error message that is not so informative:
```python
sns.stripplot(x=[], y=[], hue=[], palette=[])
```
```
<stdin>:1: FutureWarning: Passing `palette` without assigning `hue` is deprecated.
<stdin>:1: UserWarning: Ignoring `palette` because no `hue` variable has been assigned.
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File ".../seaborn/seaborn/categorical.py", line 2537, in stripplot
    p.plot_strips(
  File ".../seaborn/seaborn/categorical.py", line 265, in plot_strips
    offsets = self._nested_offsets(width, dodge)
  File ".../seaborn/seaborn/categorical.py", line 240, in _nested_offsets
    n_levels = len(self._hue_map.levels)
TypeError: object of type 'NoneType' has no len()
```
The same call using `scatterplot` results with an empty plot (as expected).
The general use case is when a Dataframe is provided as input data along with x, y and hue variables, and also a palette but the Dataframe is (unfortunately) empty. Something like:
```python
tips = sns.load_dataset("tips")
tips = tips.drop(tips.index)
sns.stripplot(data=tips, x="sex", y="tip", hue="smoker", palette=[])

**Comment by mwaskom** [comment]
I think most functions short circuit on empty x/y to avoid downstream special casing.

**Comment by mwaskom** [comment]
Closed by #3136 (although it would still be good to have these functions short circuit since otherwise you end up playing edge case whack-a-mole with empty data).
